Update SAML configuration steps (#101663)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md

- From the Identity Provider (IdP) to the Service Provider (SP):
  - `HTTP-POST` binding

### Security

> Grafana supports signed and encrypted assertions, and does _not_ support signed or encrypted requests.

Reviewer: I think this is outdated, because Grafana supports signed requests.
```suggestion
Grafana supports signed and encrypted assertions, and does _not_ support encrypted requests.
```

The SAML SSO standard uses asymmetric encryption to exchange information between the SP (Grafana) and the IdP. To perform such encryption, you need a public part and a private part. In this case, the X.509 certificate provides the public part, while the private key provides the private part. The private key needs to be issued in a [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.

{{% admonition type="note" %}}

> Directly supplying a certificate and private key to Grafana is optional. Commonly, the certificate and key are embedded in the [IDP metadata](#configure-the-saml-toolkit-application-endpoints) and refreshed as needed by Grafana automatically. This can save you the work of manually rotating your certs and keys whenever they expire.

Reviewer: > Directly supplying a certificate and private key to Grafana is optional. Commonly, the certificate and key are embedded in the [IDP metadata](#configure-the-saml-toolkit-application-endpoints)

The cert and private key that could be configured on the Grafana side are different from the certificate (and the public key) in the IDP metadata, so I think it should be rewritten to convey the message that it should be configured if their IdP expects signed requests.

{{% /admonition %}}

> Whether supplying certificate and key directly or via idp_metadata, loading certs into Grafana helps ensure that Grafana connects to a verified endpoint (such as in the case of MITM takeover scenarios).

Reviewer: This is not required, because the cert in the idp_metadata and private key (+ the certificate shared with the public key) used for request signing are different.

You can only use one form of each configuration option. Using multiple forms, such as both `certificate` and `certificate_path`, results in an error.
{{% /admonition %}}

> Always work with your company's security team on setting up certificates and private keys. If you need to generate them yourself (such as in the short term, for testing purposes, and so on), use the following example to generate your certificate and private key, including the step of ensuring that the key is generated with the [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.

Reviewer: 💯

### Example of private key generation for SAML authentication

An example of how to generate a self-signed certificate and private key that's valid for one year:

```sh
$ openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes​
```
Base64-encode the cert.pem and key.pem files:
(-w0 switch is not needed on Mac, only for Linux)

```sh
$ base64 -w0 key.pem > key.pem.base64
$ base64 -w0 cert.pem > cert.pem.base64
```
The base64-encoded values (`key.pem.base64, cert.pem.base64` files) are then used for certificate and private_key.

> Run the following command to convert the private key to [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.
> 
> ```sh
> openssl pkcs8 -topk8 -inform PEM -outform PEM -in myoriginalkey.pem -out myconvertedkey.pem -nocrypt
> ```
> 
> The base64-encoded values (`myconvertedkey.pem.base64, cert.pem.base64` files) are then used for certificate and private_key.

Reviewer: It's not needed, because the private key will be in PKCS8 format.
```suggestion
Run the following command to convert the private key to [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.

```sh
openssl pkcs8 -topk8 -inform PEM -outform PEM -in myoriginalkey.pem -out myconvertedkey.pem -nocrypt
```

The base64-encoded values (`myconvertedkey.pem.base64, cert.pem.base64` files) are then used for certificate and private_key.
```

Grafana supports user authentication through Azure AD, which is useful when you want users to access Grafana using single sign-on. This topic shows you how to configure SAML authentication in Grafana with [Azure AD](https://azure.microsoft.com/en-us/services/active-directory/).

**Before you begin**

Ensure you have permission to administer SAML authentication. For more information about roles and permissions in Grafana, refer to [Roles and permissions](/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/).

Learn the [limitations of Azure AD SAML] (https://learn.microsoft.com/en-us/entra/identity-platform/id-token-claims-reference#groups-overage-claim) integration.

> Configure SAML integration with Azure AD, [creating an app integration](https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/add-application-portal-configure) inside the Azure AD organization first.

Reviewer: I guess the link included here doesn't cover the steps for creating the integration.
```suggestion
Configure SAML integration with Azure AD, [creating an Enterprise Application](https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/add-application-portal) inside the Azure AD organization first and then [enable single sign-on](https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/add-application-portal-setup-sso).
```

> If you have users that belong to more than 150 groups, configure a registered application to provide an Azure Graph API to retrieve the groups. Refer to [Setup Azure AD Graph API applications](#set-up-saml-with-azure-ad).

Reviewer: ```suggestion
If you have users that belong to more than 150 groups, configure a registered application to provide an Azure Graph API to retrieve the groups. Refer to [Setup Azure AD Graph API applications](#configure-a-graph-api-application-in-azure-ad).
```

Grafana supports user authentication through Azure AD, which is useful when you want users to access Grafana using single sign-on. This topic shows you how to configure SAML authentication in Grafana with [Azure AD](https://azure.microsoft.com/en-us/services/active-directory/).

**Before you begin**

Ensure you have permission to administer SAML authentication. For more information about roles and permissions in Grafana, refer to [Roles and permissions](/docs/grafana/<GRAFANA_VERSION>/administration/roles-and-permissions/).

Learn the [limitations of Azure AD SAML] (https://learn.microsoft.com/en-us/entra/identity-platform/id-token-claims-reference#groups-overage-claim) integration.

> Configure SAML integration with Azure AD, [creating an app integration](https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/add-application-portal-configure)  inside the Azure AD organization first.

Reviewer: ```suggestion
Configure SAML integration with Azure AD, [creating an app integration](https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/add-application-portal-configure) inside the Azure AD organization first.
```

Starting from Grafana version 11.5, Grafana uses the `NameID` from the SAML assertion to create the logout request. If the `NameID` is not present in the assertion, Grafana defaults to using the user's `Login` attribute. Additionally, Grafana supports including the `SessionIndex` in the logout request if it is provided in the SAML assertion by the IdP.

{{% admonition type="note" %}}

> These improvements are available in public preview behind the `improvedExternalSessionHandling` feature toggle, starting from Grafana v11.5. To enable it, refer to the [Configure feature toggles](/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/feature-toggles/)

Reviewer: The correct feature toggle name should be `improvedExternalSessionHandlingSAML`. Please add this as a fix.

```suggestion
These improvements are available in public preview behind the `improvedExternalSessionHandlingSAML` feature toggle, starting from Grafana v11.5. To enable it, refer to the [Configure feature toggles](/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/feature-toggles/)
```
